app/controllers/application_metal_controller.rb

```diff
@@ -0,0 +1,10 @@
+class ApplicationMetalController < ActionController::Metal
+  # Any shared behavior across metal-oriented controllers can go here.
+
+  def session_current_user_id
+    # This method should stay in sync with the ApplicationController equivalent
+    # Could/should be extracted to the appropriate place for ideal code sharing and efficiency
```

Comment:
I like how separating this makes it very explicit. One thought, if someone goes into the ApplicationController and changes the method there without knowing about this one that would not be great. Should we add a comment to the ApplicationController in addition to this one to point out this is here?